### PR TITLE
Add url config to redis

### PR DESCRIPTION
**Pull request by pedantix**

### CODE REVIEW

**Comment by bre7** [comment]
I was going to submit a new PR but could you run `git ls-files -ci --exclude-standard -z | xargs -0 git rm --cached` to re-apply the `.gitignore` and remove the `.DS_Store` files ?

**Comment by pedantix** [comment]
@bre7 did I add a .DS_store file in here? I am not seeing it. Sorry for the late reply on this I try to be better than that.

File: Sources/Redis/Database/RedisDatabase.swift [hunk 1]
```diff
@@ -8,18 +8,22 @@ public final class RedisDatabase: Database {
     public typealias Connection = RedisClient
 
     /// This client's configuration.
-    public let config: RedisClientConfig
+    public let url: URL
```
Comment (tanner0101) [inline Sources/Redis/Database/RedisDatabase.swift]: this should still be `RedisClientConfig`

**Comment by kjcolley7** [comment]
@pedantix: Is this change intended to remove the initializer for RedisClientConfig where you give it hostname, password, etc? Just broke my project...

```
/vapor/Sources/App/configure.swift:71:20: error: cannot invoke initializer for type 'RedisClientConfig' with an argument list of type '(hostname: String, password: String?)'
        let redisConfig = RedisClientConfig(
                          ^
/vapor/Sources/App/configure.swift:71:20: note: overloads for 'RedisClientConfig' exist with these partially matching parameter lists: (url: URL), (), (from: Decoder)
        let redisConfig = RedisClientConfig(
```

File: Sources/Redis/Client/RedisClient.swift [hunk 2]
```diff
@@ -68,9 +69,32 @@ public struct RedisClientConfig: Codable {
     public var password: String?
 
     /// Create a new `RedisClientConfig`
-    public init(hostname: String = "localhost", port: Int = 6379, password: String? = nil) {
```
Comment (kjcolley7) [inline Sources/Redis/Client/RedisClient.swift]: 😢

Comment (pedantix) [reply Sources/Redis/Client/RedisClient.swift]: @kjcolley7 I don't want to break anyones project... but this is pre release :( so I am going to have to ask you to adjust as we finalize API sorry. would you be willing to try somthing like 

let hostname = "localhost"
let port = 5679
let connectionString = "redis://\(hostname):\(port)"
let url = URL(string: connectionString)
and using that URL to see if it restores your projects functionality?

Comment (kjcolley7) [reply Sources/Redis/Client/RedisClient.swift]: That works, thanks!

Comment (pedantix) [reply Sources/Redis/Client/RedisClient.swift]: thanks for using it!

**Pull request merged**